onyo/lib/commands.py

```diff
@@ -1334,20 +1326,20 @@ def onyo_unset(inventory: Inventory,
         raise ValueError("The following paths aren't assets:\n%s" % "\n".join(non_asset_paths))
     if any(k in inventory.repo.get_asset_name_keys() for k in keys):
         raise ValueError("Can't unset asset name keys.")
-    if any(k in RESERVED_KEYS + PSEUDO_KEYS for k in keys):
-        raise ValueError(f"Can't unset reserved or pseudo keys ({', '.join(RESERVED_KEYS + PSEUDO_KEYS)}).")
+    # TODO: Actual Namespaces! Key must not start with `onyo.` What about aliases?
+    if any(k in RESERVED_KEYS or k.startswith("onyo.") for k in keys):
+        raise ValueError(f"Can't unset reserved keys ({', '.join(RESERVED_KEYS)}) "
```

Comment:
I think this might be too strict.

I agree that our current pseudo-keys under `onyo.` cannot be unset. Especially since we decided that we won't allow creation/deletion through set/unset. Just modifying within the guardrails.

But the reserved keys:
```
RESERVED_KEYS = ['template', 'onyo'] + list(PSEUDOKEY_ALIASES.keys())
...
PSEUDOKEY_ALIASES: Dict[str, str] = {
    'path': 'onyo.path.relative',
    'directory': 'onyo.path.parent',
}
```

Any aliases should be translated before they hit this far into `onyo_unset()`. And if they somehow do get this far, I don't see the problem of unsetting keys that match.

And a similar argument for `template`. It should never exist in the asset content. But if it does somehow get into assets, I don't see why we should disallow unsetting it.

Reply:
Yes, this is just to not break stuff yet. Ultimately, the concept of reserved/disallowed needs to change (it's context-dependent) anyway. `onyo` is just a crude measure to account for the top-level key and `template` is supposed to become `onyo.cli.template` and would then not be part of the dict indeed.

So, in short: Agree, this needs to change, but only after there's a better way to specify namespaces (`onyo.` in this case) and `template` has moved to where it now belongs.